starter-code/InvestingPortfolio/app/src/main/java/drewmahrt/generalassemb/ly/investingportfolio/MainActivity.java

```diff
@@ -4,49 +4,144 @@
 import android.content.ContentValues;
 import android.content.DialogInterface;
 import android.database.Cursor;
+import android.net.Uri;
 import android.os.Bundle;
 import android.support.v4.app.LoaderManager;
 import android.support.v4.content.CursorLoader;
 import android.support.v4.content.Loader;
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
 import android.support.v7.widget.Toolbar;
 import android.util.Log;
-import android.view.LayoutInflater;
+import android.view.View;
 import android.widget.EditText;
-import android.widget.Toast;
-
-import com.android.volley.Request;
-import com.android.volley.RequestQueue;
-import com.android.volley.Response;
-import com.android.volley.VolleyError;
-import com.android.volley.toolbox.JsonArrayRequest;
-import com.android.volley.toolbox.JsonObjectRequest;
-import com.android.volley.toolbox.Volley;
 
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+
 public class MainActivity extends AppCompatActivity implements LoaderManager.LoaderCallbacks<Cursor>{
     private static final String TAG = "MainActivity";
     public static final int LOADER_STOCK = 0;
+    public static final Uri CONTENT_URI = StockPortfolioContract.Stocks.CONTENT_URI;
+    public static final String BASE_URL = "http://dev.markitondemand.com/Api/v2/lookup/json?input=";
 
     RecyclerView mPortfolioRecyclerView;
     StockRecyclerViewAdapter mAdapter;
 
+    private ContentResolver mContentResolver;
+
+    private List<Stock> mStocks;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
+        mContentResolver = getContentResolver();
+
+        mStocks = new ArrayList<>();
+
         //TODO: Initialize StockRecyclerViewAdapter
+        mPortfolioRecyclerView = (RecyclerView)findViewById(R.id.portfolio_list);
+        mPortfolioRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+
+        mAdapter = new StockRecyclerViewAdapter(mStocks);
+        mPortfolioRecyclerView.setAdapter(mAdapter);
+
 
         getSupportLoaderManager().initLoader(LOADER_STOCK,null,this);
 
+
+        findViewById(R.id.fab).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
```

Comment:
Might want to consider using a dialog fragment to help move some code to another file and make it easier to read